### PR TITLE
Add workflow for update api spec files to main

### PR DESCRIPTION
**Pull request by pedro-psb**
I've added this data-update workflow on the docs-update branch and merged it (https://github.com/pulp/pulp-docs/pull/66).

This workflow should really be on the default branch to work.

### CODE REVIEW

File: .github/workflows/update-data.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+          ref: docs-data
+
+      - name: "Set up Python"
+        uses: "actions/setup-python@v5"
+        with:
+          python-version: "3.11"
+
+      - name: "List requirements"
+        run: |
+          cat requirements_system.txt
```
Comment (mdellweg) [inline .github/workflows/update-data.yml]: Is it system or ubuntu?

File: .github/workflows/update-data.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          cat requirements_python.txt
+
+      - name: "Install System dependencies"
+        run: |
+          xargs sudo apt -y install < requirements_ubuntu.txt
```
Comment (mdellweg) [inline .github/workflows/update-data.yml]: Ideally we call apt install only once, but xargs will at least do batches, right?

File: .github/workflows/update-data.yml (hunk 1 above)
Comment (pedro-psb) [reply .github/workflows/update-data.yml]: its ubuntu, thanks

File: .github/workflows/update-data.yml (hunk 2 above)
Comment (pedro-psb) [reply .github/workflows/update-data.yml]: Not sure. Tried a different approach with `tr`.

**Review by mdellweg — APPROVED** [review]
Thanks fro trying to avoid all the yaml weirdness with quotes.

**Pull request merged**